### PR TITLE
Add WithErrorHandler client option

### PR DESCRIPTION
**Pull request by jokly**
Added the ability to change the error handler for `retryablehttp.Client`

### CODE REVIEW

File: client_options.go [hunk 1]
```diff
@@ -118,6 +118,14 @@ func WithResponseLogHook(hook retryablehttp.ResponseLogHook) ClientOptionFunc {
 	}
 }
 
+// WithErrorHandler can be used to configure a custom error handler.
+func WithErrorHandler(handler retryablehttp.ErrorHandler) ClientOptionFunc {
+	return func(c *Client) error {
+		c.client.ErrorHandler = handler
+		return nil
+	}
+}
```
Comment (svanharmelen) [inline client_options.go:127]: These options are ordered alphabetically, so if you can update the location of the new option that would be great...

**Review by svanharmelen — CHANGES_REQUESTED** [review]
One small request, after that is updated I'll merge this one 👍🏻

File: client_options.go (hunk 1 above)
Comment (jokly) [reply client_options.go]: Thanks for the quick response! Fixed this :)

**Review by svanharmelen — APPROVED** [review]
Thanks 👍🏻

**Pull request merged**